### PR TITLE
Add components

### PR DESCRIPTION
**Pull request by bohdan-meltiukhov**
# UI Components
This PR adds markup created using components. It includes:
- An abstract component that can render the content;
- The component for the application, which contains the login form;
- The login form component that initializes and places inputs, buttons, headers and footers,
- Components for different parts of the page, such as button, input, header and footer.

### CODE REVIEW

**Comment by dkalinin** [comment]
Please add description.

File: ui-components/tests/components/form-input/index.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    const input = formInput.querySelector('.input');
+    assert.strictEqual(input.placeholder, 'Placeholder', 'The form input should have the default placeholder.');
+
+    assert.strictEqual(input.type, 'text', 'The form input should have the default type.');
+});
+
+test('should display the provided properties.', (assert) => {
+    const labelText = 'Password';
+    const placeholder = 'Password';
```
Comment (dkalinin) [inline ui-components/tests/components/form-input/index.js]: Let's have different string for `labelText` and `placeholder`.

File: ui-components/app/components/form-input/index.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+export default class FormInput extends Component {
+
+    constructor(container, {labelText = 'Label', placeholder = 'Placeholder', type = 'text'} = {}) {
+        super(container);
+
+        this._labelText = labelText;
+        this._placeholder = placeholder;
+        this._type = type;
+        this._helpText = '';
```
[the remark this answers is not in the document]
Comment (bohdan-meltiukhov) [reply ui-components/app/components/form-input/index.js]: I removed the `_helpText` field as redundant.

File: ui-components/app/components/login-form-footer/index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import Component from "../component.js";
+import Button from "../button";
+
+export default class FormFooter extends Component {
+
+    constructor(container, {
+        buttonText = 'Submit',
+        linkText = 'Take me to another page',
+        linkDirection = '#'
```
[the remark this answers is not in the document]
Comment (bohdan-meltiukhov) [reply ui-components/app/components/login-form-footer/index.js]: I renamed it.

File: ui-components/tests/components/form-input/index.js (hunk 1 above)
Comment (bohdan-meltiukhov) [reply ui-components/tests/components/form-input/index.js]: Done.

**Pull request merged**